server/src/lib.rs

```diff
@@ -161,7 +161,12 @@ impl Server {
             ClientEvent::Hello(hello) => {
                 assert!(client.handles.is_none());
                 let snapshot = Arc::new(self.sim.snapshot());
-                let (id, entity) = self.sim.spawn_character(hello);
+                let Some((id, entity)) = self.sim.activate_or_spawn_character(&hello) else {
+                    error!("could not spawn {} due to name conflict", hello.name);
+                    client.conn.close(3u32.into(), b"name conflict");
```

Comment:
"3" is an opaque magic number. Introduce a named constant in common::proto instead.

Reply:
I went ahead and replaced the other opaque magic numbers used the same way as well.